core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java

```diff
@@ -33,23 +33,25 @@
   @Override
   public void accept(FluxSink<T> fluxSink) {
     FluxSinkRecorderDelegate<T> previousDelegate = this.delegate;
-    delegate = new DirectDelegate<>(fluxSink);
-    previousDelegate.accept(fluxSink);
+    synchronized (this) {
+      delegate = new DirectDelegate<>(fluxSink);
+      previousDelegate.accept(fluxSink);
```

Comment:
shouldn't this be outside the critical region?

Reply:
This is the change that prevents the race conditon.